api/src/main/java/org/openmrs/module/fhir2/api/FhirObservationService.java

```diff
@@ -23,5 +26,7 @@
 	Observation getObservationByUuid(@NotNull String uuid);
 	
 	Collection<Observation> searchForObservations(ReferenceParam encounterReference, ReferenceParam patientReference,
+	        ReferenceParam hasMemberReference, TokenAndListParam valueConcept, DateRangeParam valueDateParam,
```

Comment:
@ibacher I have a few questions here
1. What should be the data type of hasMemberReference I think that `TokenOrListParam` should be the datatype as we do have a function in `BaseDaoImpl` to handleBoolean.
2. I am a bit confused that how should we handle the `value` parameters, should we have only a single ReferenceParam and include all of these value-concept value-date value-quantity value-string, and then make a function to handle all the value related search queries.

Reply:
@VaishSiddharth

1. So the `hasMember` of [FHIR Observations](https://www.hl7.org/fhir/observation-definitions.html#Observation.hasMember) is not a boolean value. It stores a reference to an observation that is part of this observation group. So this *should* be a `ReferenceParam` and treated similarly to how we treat `Subject`, etc.

2. The [FHIR spec](https://www.hl7.org/fhir/observation.html#search) actually wants separate parameters for each value type, so `value-concept` should map to a `TokenAndListParam`, `value-date` to a `DateRangeParam`, `value-quantity` to a `QuantityAndListParam` and `value-string` to a `StringAndListParam`.